src/conn/conn_cache.c

```diff
@@ -46,6 +46,111 @@ __cache_config_abs_to_pct(
     return (0);
 }
 
+/*
+ * __validate_cache_config --
+ *     Validate trigger and target values of given configs.
+ */
+static int
+__validate_cache_config(WT_SESSION_IMPL *session, const char *cfg[], bool create)
+{
+    WT_CACHE *cache;
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    bool debug_config;
+
+    conn = S2C(session);
+    cache = conn->cache;
+
+    WT_RET(__wt_config_gets(session, cfg, "debug_mode.configuration", &cval));
+    debug_config = cval.val;
+
+    /* The target size must be lower than the trigger size or we will never get any work done. */
+    if (create) {
+        if (cache->eviction_target >= cache->eviction_trigger)
+            WT_RET_MSG(session, EINVAL, "eviction target must be lower than the eviction trigger");
+        if (cache->eviction_dirty_target >= cache->eviction_dirty_trigger)
+            WT_RET_MSG(session, EINVAL,
+              "eviction dirty target must be lower than the eviction dirty trigger");
+        if (cache->eviction_updates_target >= cache->eviction_updates_trigger)
+            WT_RET_MSG(session, EINVAL,
+              "eviction updates target must be lower than the eviction updates trigger");
+        return (0);
+    }
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_target", &cval));
+    cache->eviction_target = (double)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_trigger", &cval));
+    cache->eviction_trigger = (double)cval.val;
```

Comment:
If someone can reconfigure all the values, don't we need to repeat all the same invalid checks we did on `create`? For example, if one can reconfigure both `cache->eviction_target` and `cache->eviction_trigger` don't we need to check like at line 69? The checks below do not repeat the checks above but do potentially change values in an invalid way.

If so, then this code should be re-arranged to have one set of those EINVAL conditionals.

Reply:
I agree that both checks should be performed, but using EINVAL for all checks causes a lot of failures, I will look into finding an approach to log these messages without breaking our tests.